python/whylogs/core/view/segmented_dataset_profile_view.py

```diff
@@ -0,0 +1,184 @@
+        metric_name_indices: Dict[str, int] = {}
+        metric_index_to_name: Dict[int, str] = {}
+        for i in range(0, len(metric_name_list)):
+            metric_name_indices[metric_name_list[i]] = i
+            metric_index_to_name[i] = metric_name_list[i]
+
+        column_chunk_offsets: Dict[str, ChunkOffsets] = {}
+        with tempfile.TemporaryFile("w+b") as f:
+            for col_name in sorted(self.profile_view._columns.keys()):
+                column_chunk_offsets[col_name] = ChunkOffsets(offsets=[f.tell()])
+
+                col = self.profile_view._columns[col_name]
+
+                # for a given column, turn it into a ChunkMessage.
+                indexed_component_messages: Dict[int, MetricComponentMessage] = {}
+                metric_components = col.to_protobuf().metric_components
+                for m_name, m_component in metric_components.items():
+                    index = metric_name_indices.get(m_name)
+                    if index is None:
+                        raise ValueError(f"Missing metric from index mapping. Metric name: {m_name}")
+                    indexed_component_messages[index] = m_component
+
+                chunk_msg = ChunkMessage(metric_components=indexed_component_messages)
+                chunk_header = ChunkHeader(type=ChunkHeader.ChunkType.COLUMN, length=chunk_msg.ByteSize())
+                write_delimited_protobuf(f, chunk_header)
+                f.write(chunk_msg.SerializeToString())
+
+            total_len = f.tell()
+            f.flush()
+
+            # calculate segment tags based on columnar segments
+            segment_message_tags, segment_tags, segment_metadata = _generate_segment_tags_metadata(
+                self.segment, self.partition
+            )
+
+            if not segment_tags:
+                raise NotImplementedError(
+                    f"Serialization of segments requires segment tags but none calculated for partition: {self.partition} and segments: {self.segment}"
+                )
+
+            segment_message = SegmentMessage()
+            segment_message.tags.extend(segment_tags)
+            segments_message_field = [segment_message]
+
+            properties = DatasetProperties(
+                dataset_timestamp=to_utc_milliseconds(self.profile_view._dataset_timestamp),
+                creation_timestamp=to_utc_milliseconds(self.profile_view._creation_timestamp),
+                metadata=segment_metadata,
+                tags=segment_message_tags,
+            )
+
+            logger.warn(f"constructed DatasetProperties for segmented profile file: {properties}")
+
+            dataset_header = DatasetProfileHeader(
+                column_offsets=column_chunk_offsets,
+                properties=properties,
+                length=total_len,
+                indexed_metric_paths=metric_index_to_name,
+            )
+
+            # TODO: multi segment file format requires multiple offset calculations.
```

Comment:
Is this TODO: still valid? Are we just oding single file segments right now?

Reply:
correct just single segment files for v1 initial

Reply:
Updated comment.